src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -71,16 +80,20 @@ private void processBlob(BlobClient blob) {
         envelopeService
             .findEnvelope(blob.getBlobName(), blob.getContainerName())
             .ifPresentOrElse(
-                envelope -> logger.info(
-                    "Envelope already processed in system, skipping."
-                        + " ID: {}, filename: {}, container: {}, state: {}",
-                    envelope.id,
-                    envelope.fileName,
-                    envelope.container,
-                    envelope.status.name()
-                ),
+                envelope -> {
+                    if (isStale(envelope)) {
+                        logger.warn("Found stale envelope. Resuming its processing. {}", envelope.getBasicInfo());
+                        blobProcessor.continueProcessing(envelope.id, blob);
+                    } else {
+                        logger.info("Envelope already processed in system, skipping. {}", envelope.getBasicInfo());
```

Comment:
If the envelope is created but not old enough, this message will be misleading.

What problem does 'stale' solve?

Reply:
good point.

waiting for 5min to not proceed with envelope that is currently being processed for the first time

Reply:
Ok, may save us some unnecessary log entries.